ecl/hthor/hthorkey.cpp

```diff
@@ -4187,11 +4188,17 @@ class CHThorKeyedJoinActivity  : public CHThorThreadedActivityBase, implements I
         if (actualFormat)
         {
             actualLayouts.append(actualFormat.getLink());  // ensure adequate lifespan
-            Owned<const IDynamicTransform> payloadTranslator =  createRecordTranslator(helper.queryProjectedIndexRecordSize()->queryRecordAccessor(true), actualFormat->queryRecordAccessor(true));
+            Owned<const IDynamicTransform> payloadTranslator = createRecordTranslator(helper.queryProjectedIndexRecordSize()->queryRecordAccessor(true), actualFormat->queryRecordAccessor(true));
+            DBGLOG("Record layout translator created for %s", f->queryLogicalName());
```

Comment:
What is the rule for when logging should appear, and the translator should be described?  I imagine the output might be quite large.  Does it differ for files and keys?

Reply:
Should it only be if expected != actual, rather than projected != expected?

Reply:
To have reached this code, actual and expected must differ (we don't have the three layouts in indexes yet)